packages/lesswrong/components/posts/BookmarksPage.jsx

```diff
@@ -2,18 +2,27 @@ import { registerComponent, Components } from 'meteor/vulcan:core';
 import React from 'react';
 import withErrorBoundary from '../common/withErrorBoundary';
 import {AnalyticsContext} from "../../lib/analyticsEvents";
+import {useCurrentUser} from "../common/withUser"
 
 const BookmarksPage = () => {
   const { SingleColumnSection, SectionTitle, BookmarksList} = Components
 
-  return (
-    <SingleColumnSection>
-        <AnalyticsContext listContext={"bookmarksPage"} capturePostItemOnMount>
-            <SectionTitle title="Bookmarks"/>
-            <BookmarksList />
-        </AnalyticsContext>
-    </SingleColumnSection>
-  )
+  const currentUser = useCurrentUser()
+
```

Comment:
Instead of if/then, do a single "if (!currentUser) return <span>You must sign in to view bookmarked posts.</span>"